custom/icds_reports/utils/aggregation_helpers/distributed/agg_dashboard_acitivity.py

```diff
@@ -0,0 +1,225 @@
+from datetime import timedelta
+from django.db.models import Max
+
+import re
+
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import BaseICDSAggregationDistributedHelper
+from custom.icds_reports.const import  AGG_DASHBOARD_ACTIVITY
+from corehq.apps.users.dbaccessors.all_commcare_users import get_all_user_id_username_pairs_by_domain
+from django.utils.functional import cached_property
+from corehq.apps.users.models import CommCareUser
+
+
+class DashboardActivityReportAggregate(BaseICDSAggregationDistributedHelper):
+    aggregate_parent_table = AGG_DASHBOARD_ACTIVITY
+
+    def __init__(self, date):
+        self.date = date
+        self.last_agg_date = self.get_last_agg_date()
+
+    def aggregate(self, cursor):
+        cursor.execute(self.drop_table_query())
+        cursor.execute(*self.create_table_query())
+        add_query, add_params = self.add_missing_users()
+
+        cursor.executemany(add_query, add_params['values'])
+
+        cursor.execute(*self.rollover_previous_data())
+
+        update_queries = self.update_query()
+
+        for query, param in update_queries:
+            cursor.execute(query, param)
+
+    @cached_property
+    def get_dashboard_users(self):
+        all_users = get_all_user_id_username_pairs_by_domain(self.domain, include_web_users=False,
+                                                             include_mobile_users=True)
+        dashboard_uname_rx = re.compile(r'^\d*\.[a-zA-Z]*@.*')
```

Comment:
This is going to retrieve 1.4 million users at scale. Is there anyway you can limit the number of users it will return from the database?

Reply:
I could not find any method to pull just a smaller chunk. I will again check tomorrow.

Reply:
@emord  @calellowitz sory. how will it be 1.4 million? I found there are about 700K users only. Do you think that 700K is also going to be a problem.

Reply:
nvm. I changed the code to pull the limited numbers of users. now only required users are pulled i.e 18K